servers/src/common/types.rs

```diff
@@ -113,6 +114,12 @@ pub struct ServerConfig {
 	/// Network address for the Rest API HTTP server.
 	pub api_http_addr: String,
 
+	/// Whether to enable basic access authentication on API
+	pub api_basic_auth: Option<bool>,
+
+	/// Secret for basic auth on Rest API HTTP server.
+	pub api_secret: String,
```

Comment:
Is this something that is going to get written down to the grin server config? I would urge this to be a path to the actual secret vs mixing credentials with config.

Reply:
You mean using environment variable for example or something else?

Reply:
Not an env variable but a path to a different file (similar to `db_root` or `log_file_path`) where you store the secret. The reason for doing this is to avoid treating the main config file as a secret blob (you need to if you start adding sensitive information to it).

Reply:
Okay I see, let me see if I can add this here.